### PR TITLE
Add Shippable support for testing the web ui

### PR DESCRIPTION
**Pull request by bcbroussard**
Add protractor tests to shippable, as described in https://github.com/GoogleCloudPlatform/kubernetes/issues/7122. 

This PR also depends on https://github.com/GoogleCloudPlatform/kubernetes/pull/8639 

/cc @jackgr @lavalamp @bgrant0607

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -26,7 +26,13 @@ install:
   - PATH=$HOME/gopath/bin:./third_party/etcd:$PATH ./hack/verify-gendocs.sh
 
 before_script:
-  - npm install karma karma-junit-reporter karma-phantomjs-launcher karma-jasmine
+  - travis_retry npm install karma@~0.10 karma-phantomjs-launcher@~0.1.4 karma-cli@0.0.4 karma-jasmine@^0.1.5 karma-junit-reporter@^0.2.2 karma-story-reporter@^0.3.1 protractor@^2.1.0
```
Comment (lavalamp) [inline .travis.yml]: Can you package this stuff up in a script and call that? That'd be much clearer (and easier to try locally if desired).

File: shippable.yml [hunk 2]
```diff
@@ -6,9 +6,9 @@ build_image: shipimg/ubuntu1204_go:latest
 
 matrix:
   include:
-    - go: 1.4
-      env:
-        - KUBE_TEST_API_VERSIONS=v1beta1 KUBE_TEST_ETCD_PREFIXES=registry
+    # - go: 1.4
+    #   env:
+    #     - KUBE_TEST_API_VERSIONS=v1beta1 KUBE_TEST_ETCD_PREFIXES=registry
```
Comment (lavalamp) [inline shippable.yml]: Why did this get commented out?

File: shippable.yml [hunk 3]
```diff
@@ -31,12 +35,24 @@ install:
...
+  - sleep 3 # give xvfb some time to start
 
 script:
-  - KUBE_RACE="-race" KUBE_COVER="y" KUBE_GOVERALLS_BIN="$HOME/gopath/bin/goveralls" KUBE_TIMEOUT='-timeout 300s' KUBE_COVERPROCS=8 KUBE_TEST_ETCD_PREFIXES="${KUBE_TEST_ETCD_PREFIXES}" KUBE_TEST_API_VERSIONS="${KUBE_TEST_API_VERSIONS}" ./hack/test-go.sh -- -p=2
-  - PATH=$HOME/gopath/bin:./third_party/etcd:$PATH ./hack/test-cmd.sh
-  - PATH=$HOME/gopath/bin:./third_party/etcd:$PATH KUBE_TEST_API_VERSIONS="${KUBE_TEST_API_VERSIONS}" KUBE_INTEGRATION_TEST_MAX_CONCURRENCY=4 LOG_LEVEL=4 ./hack/test-integration.sh
-  - PATH=$HOME/gopath/bin:./third_party/etcd:$PATH ./hack/test-update-storage-objects.sh
+  # - KUBE_RACE="-race" KUBE_COVER="y" KUBE_GOVERALLS_BIN="$HOME/gopath/bin/goveralls" KUBE_TIMEOUT='-timeout 300s' KUBE_COVERPROCS=8 KUBE_TEST_ETCD_PREFIXES="${KUBE_TEST_ETCD_PREFIXES}" KUBE_TEST_API_VERSIONS="${KUBE_TEST_API_VERSIONS}" ./hack/test-go.sh -- -p=2
+  # - PATH=$HOME/gopath/bin:./third_party/etcd:$PATH ./hack/test-cmd.sh
+  # - PATH=$HOME/gopath/bin:./third_party/etcd:$PATH KUBE_TEST_API_VERSIONS="${KUBE_TEST_API_VERSIONS}" KUBE_INTEGRATION_TEST_MAX_CONCURRENCY=4 LOG_LEVEL=4 ./hack/test-integration.sh
```
Comment (lavalamp) [inline shippable.yml]: Oh, you're commenting out for testing? tag me here when you get it working & I'll review.

**Comment by bcbroussard** [comment]
@lavalamp yes, still a WIP with testing comments. I'll ping you later today when this is wrapped up.

**Comment by bcbroussard** [comment]
@kprobst @lavalamp - This is ready for review. I'll rebase and update the api to v1 if https://github.com/GoogleCloudPlatform/kubernetes/pull/9400 goes in ahead of this.

File: .travis.yml (hunk 1 above)
Comment (bcbroussard) [inline .travis.yml]: @lavalamp this is actually part of the Travis updates PR https://github.com/GoogleCloudPlatform/kubernetes/pull/8639 . 

The dependencies are defined in npm's package.json (`www/master/package.json`) for local testing. Travis and Shippable have a small subset of dependencies defined here to run the browser based tests in a headless environment, or with a natively installed browser (firefox).

**Comment by kprobst** [comment]
It looks like everything in this PR is already in PR #8639 (which you mention in you PR description as a dependency). 

According to https://github.com/bcbroussard/kubernetes/compare/shippable-protractor...bcbroussard:travis-protractor the travis-protactor and shippable-protractor branches are the same.

What am I missing?

**Comment by bcbroussard** [comment]
@kprobst I think reversing the order will give you what you are looking for - https://github.com/bcbroussard/kubernetes/compare/travis-protractor...bcbroussard:shippable-protractor

The first commit is for the travis updates since there is shared testing code between travis and shippable.

**Comment by kprobst** [comment]
Mostly lg, but I agree with lxdy that it would be cleaner to have the shippable setup be part of a separate script rather than it be part of hack/test-web-e2e.sh. This would also create more consistency between the travis and shippable scripts/ymls.

**Comment by ixdy** [comment]
I left a number of comments in #8639 that are still applicable here.

**Comment by ixdy** [comment]
ping @bcbroussard - are you still interested in merging this PR?

**Comment by wojtek-t** [comment]
@bcbroussard - ping

**Pull request closed without merging**